x-pack/plugins/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -164,8 +250,9 @@ export class KnowledgeBaseService {
       return;
     }
 
+    this._queue.push(...operations);
```

Comment:
why add this to the queue if you're going to process them anyway?

Reply:
I would imagine having it in the queue would help deal if `pRetry` fails for some reason?

Reply:
There is no pRetry I think, just pLimit.

Reply:
Ok.

Note to self: need to implement chunking

Reply:
Are you saying I should not use the `queue` method on the kb service for importing at all?

Will that not block the event loop in the case of a lot of objects?

Reply:
Yeah the queue is for a background process, but that's not related to the event loop, but to the fact that the knowledge base might become available in the background (because the user has enabled it). I wouldn't worry too much about event loop delay, that already happens when parsing the file send to the API endpoint. The rest is mostly ES.